### PR TITLE
Add `contract alias remove`.

### PR DESCRIPTION
**Pull request by fnando**
### What

Add command to remove contract aliases.

```console
$ cat ~/.config/soroban/contract-ids/increment.json
{"ids":{"Test SDF Network ; September 2015":"CCYBGMETD5TYZJ2ZBT3JBBCWODYJJWP274JA4LX4S7JGNUNQPXZBMNUY"}}%

$ cargo run contract alias remove increment --network standalone --global
   Compiling soroban-cli v21.4.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/soroban-cli)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3.22s
     Running `target/debug/soroban contract alias remove increment --network standalone --global`
ℹ️ Network passphrase: Standalone Network ; February 2017
error: no contract found with alias `increment`

$ cargo run contract alias remove increment --network testnet --global
   Compiling soroban-cli v21.4.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/soroban-cli)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3.01s
     Running `target/debug/soroban contract alias remove increment --network testnet --global`
ℹ️ Network passphrase: Test SDF Network ; September 2015
ℹ️ Contract is CCYBGMETD5TYZJ2ZBT3JBBCWODYJJWP274JA4LX4S7JGNUNQPXZBMNUY
✅ Contract alias has been removed

$ cat ~/.config/soroban/contract-ids/increment.json
{"ids":{}}%
```

### Why

https://github.com/stellar/stellar-cli/issues/1420

### Known limitations

N/A

### CODE REVIEW

File: cmd/soroban-cli/src/commands/contract/alias/remove.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+use crate::commands::{config::network, global};
+use crate::config::locator;
+use crate::print::Print;
+
+#[derive(Parser, Debug, Clone)]
+// #[command(group(
+//     clap::ArgGroup::new("wasm_src")
+//         .required(true)
+//         .args(&["wasm", "wasm_hash"]),
+// ))]
```
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/contract/alias/remove.rs:14]: ❓

File: cmd/soroban-cli/src/commands/contract/alias/remove.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+impl Cmd {
+    #[allow(clippy::unused_async)]
+    pub async fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+        let alias = &self.alias;
+        let network = self.network.get(&self.config_locator)?;
+
+        print.infoln(format!(
+            "Network passphrase: {passphrase}",
+            passphrase = network.network_passphrase
```
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/contract/alias/remove.rs:48]: It feels random that this gets printed out. As a user I'm not sure what it means. I think we need to tell the narrative a little to take the user on the journey about how these things relate:

```
ℹ️ Contract alias '<alias>' references CCYBGMETD5TYZJ2ZBT3JBBCWODYJJWP274JA4LX4S7JGNUNQPXZBMNUY on network 'Test SDF Network ; September 2015'
✅ Contract alias '<alias>' has been removed
```

File: cmd/soroban-cli/src/commands/contract/alias/remove.rs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            .config_locator
+            .get_contract_id(&self.alias, &network.network_passphrase)?;
+
+        if contract.is_none() {
+            return Err(Error::NoContract {
+                alias: alias.into(),
+            });
+        };
+
+        let contract = contract.expect("contract must be set");
```
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/contract/alias/remove.rs:61]: We've had bugs arise from the use of expect/unwrap vs safe unwrapping. Safe unwrapping is a small modification on this flow:
```suggestion
        let Some(contract) = contract else {
            return Err(Error::NoContract {
                alias: alias.into(),
            });
        };
```

**Review by leighmcculloch — COMMENTED** [review]
Couple requests inline.

File: cmd/soroban-cli/src/commands/contract/alias/remove.rs (hunk 1 above)
Comment (fnando) [reply cmd/soroban-cli/src/commands/contract/alias/remove.rs:14]: :hide:

**Review by leighmcculloch — APPROVED** [review]

File: cmd/soroban-cli/src/commands/contract/alias/remove.rs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        };
+
+        print.infoln(format!(
+            "Contract alias '{alias}' references {contract} on network '{network_passphrase}'"
+        ));
+
+        self.config_locator
+            .remove_contract_id(&network.network_passphrase, alias)?;
+
+        print.checkln(format!("Contract alias '{alias}' has been removed"));
```
Comment (leighmcculloch) [inline cmd/soroban-cli/src/commands/contract/alias/remove.rs:58]: "has been" is unnecessary
```suggestion
        print.checkln(format!("Contract alias '{alias}' removed"));
```

**Comment by langer933** [comment]
Fdff

**Pull request merged**